client/lib/catch-js-errors/index.js

```diff
@@ -11,6 +11,18 @@ export default class ErrorLogger {
 
 		if ( ! window.onerror ) {
 			TraceKit.report.subscribe( errorReport => {
+				if ( Array.isArray( errorReport.stack ) ) {
+					errorReport.stack.forEach( report => Object.keys( report ).forEach( key => {
+						if ( key === 'context' ) {
+							report[ key ] = JSON.stringify( report[ key ] ).substring( 0, 512 );
```

Comment:
What if `JSON.stringify( report[ key ] )` returns `undefined`? `substring` won't work and will produce error then.